packages/golden-layout/src/items/Stack.ts

```diff
@@ -31,47 +35,33 @@ type ContentAreaDimensions = {
 
 type BodySegment = keyof ContentAreaDimensions;
 
-export interface StackHeaderConfig {
-  show?: boolean | 'top' | 'left' | 'right' | 'bottom';
-  popout?: string;
-  maximise?: string;
-  close?: string;
-  minimise?: string;
-}
-
 export default class Stack extends AbstractContentItem {
   private _activeContentItem: AbstractContentItem | null = null;
 
-  _header: StackHeaderConfig;
+  _header: StackItemHeaderConfig;
 
   childElementContainer = $('<div class="lm_items"></div>');
   header: Header;
-  parent: RowOrColumn;
+  parent: AbstractContentItem | null;
 
   isStack = true;
 
-  private _dropZones = {};
   private _dropSegment: string | null = null;
   _contentAreaDimensions: ContentAreaDimensions | null = null;
   private _dropIndex: number | undefined;
   _side: boolean | 'top' | 'left' | 'right' | 'bottom';
   _sided: boolean = false;
 
-  config: ComponentConfig & {
-    activeItemIndex?: number;
-  };
+  config: StackItemConfig;
 
   constructor(
     layoutManager: LayoutManager & {
       config: LayoutManager['config'] & {
-        header?: StackHeaderConfig;
+        header?: StackItemHeaderConfig;
       };
     },
-    config: ComponentConfig & {
-      header?: StackHeaderConfig;
-      hasHeaders?: boolean;
-    },
-    parent: RowOrColumn
+    config: StackItemConfig,
+    parent: AbstractContentItem | null
```

Comment:
I inferred that this needs to be `AbstractContentItem` instead of `RowOrColumn` based on the fact that `Stack` calls `this.layoutManager.createContentItem` passing itself as the parent. We seem to do some checking later to identify if it is a valid row or column item and if not creating one, so seemed ok to broaden the type here.

Reply:
Yea in the examples they do show nested stacks: https://golden-layout.com/examples/#b40c19e79915493e9d3ae0220773358f
So a Stack could be in a Stack. That being said, that example above seems to have bugs as soon as you start moving things around. But I digress, seems fine for these changes here.